@here/harp-texture-atlas-generator/src/AtlasGenerator.ts

```diff
@@ -0,0 +1,418 @@
+    return storeImages(inputFiles, images, outputDir, outputFormat);
+}
+
+function processImages(inputFiles: string[], imageFactory: ImageFactory,
+    operations: ImageProcess[], logger: Logger, jobsNum: number = 8,
+    outputDir?: string): Promise<ImageEncoder[]> {
+
+    // Process the input images in the source folder.
+    // Since it will run in one process per image, the number of parallel jobs (promises)
+    // is limited by options parmeter 'jobs'. This should be choosen wiselly, such like
+    // 2 jobs per single CPU core. Otherwise, the memory usage may explode (> 16GB).
+    const readPromises: Array<Promise<ImageEncoder>> = [];
+
+    // TODO: Solve imports
+    const promiseLimit = require("promise-limit");
+    const jobLimit = promiseLimit(jobsNum);
+
+    // Create image post-processing promises.
+    const outputExt: string = ".png";
+    for (const inputFile of inputFiles) {
+        let outFile: string;
+        if (outputDir !== undefined) {
+            outFile = path.join(
+                outputDir,
+                path.basename(inputFile, path.extname(inputFile)) + outputExt
+            );
+        }
+        const readPromise: Promise<ImageEncoder> = jobLimit(() => {
+            return readAndProcessImage(inputFile, imageFactory, operations, logger, outFile);
+        });
+        readPromises.push(readPromise);
+    }
+
+    // Process all input images in parallel jobs.
+    return Promise.all(readPromises);
+}
+
+async function storeImages(inputFiles: string[], images: ImageEncoder[],
+    outputDir: string, outputFormat: ImageFormat = ImageFormat.PNG): Promise<string[]> {
+
+    if (inputFiles.length !== images.length) {
+        throw new Error("Not all files has been processed, files and encoders lists vary in size!");
+    }
+
+    const outputExt: string = FileSystem.getImageFormatExtensions(outputFormat)[0];
+    const outputFiles: string[] = [];
+    for (let i = 0; i < inputFiles.length; ++i) {
+        const inputFile: string = inputFiles[i];
+        const encoder: ImageEncoder = images[i];
+        const outFile: string = path.join(
+            outputDir,
+            path.basename(inputFile, path.extname(inputFile)) + outputExt
+        );
+        await encoder.write(outFile);
+        outputFiles.push(outFile);
+    }
+    return outputFiles;
+}
+
+function mergeSpritesIntoAtlas(inputFiles: string[], outputPath: string,
+    paddingSize: number, minify: boolean, logger: Logger): Promise<string> {
+
+    // Generate texture atlas using spritesmith tool.
+    const atlasPath = outputPath + ".png";
+    const descriptorPath = outputPath + ".json";
+    const spritesmith = require("spritesmith");
+    return new Promise((resolve, reject) => {
+        spritesmith.run({ src: inputFiles, padding: paddingSize },
+            (err: Error, result: any) => {
+                if (err) {
+                    reject(new Error("Atlas creation failed! " + err));
```

Comment:
Add `return` after `reject`.
`reject` doesn't return and rest of your code will run ...

Reply:
Good point!